src/repart/repart.c

```diff
@@ -7486,6 +7490,19 @@ static int context_fstab(Context *context) {
                 }
         }
 
+        if (arg_merge_fstab) {
+                _cleanup_free_ char *h = NULL;
+
+                r = read_one_line_file(path, &h);
+                if (r > 0 && strcmp(h, fstab_hdr) != 0) {
+                        r = read_full_file(path, &c, &l);
+                        if (r >= 0 && l > 0) {
+                                fprintf(f, "%s", c);
```

Comment:
this is a bit sloppy because you read the file as binary (i.e. you ask for the size to be retruned by passing the third arg to read_full_file() as non-NULL), but then output it here as regular string.

just do fwrite() instead and pass the size?

Reply:
Done. The new version would be more clear with fprintf the return of strstr, but is using fwrite now.